homeassistant/components/homekit_controller/connection.py

```diff
@@ -139,94 +127,62 @@ def refresh_entity_map(self, config_num):
         # aid/iid to GATT characteristics. So push it to there as well.
         self.pairing.pairing_data['accessories'] = self.accessories
 
+        self.async_load_platforms()
+
         # Register add new entities that are available
-        await self.hass.async_add_executor_job(self.add_entities)
+        self.add_entities()
```

Comment:
Why do we call this here? Aren't entities added as part of loading the platforms, in `async_load_platforms` which forwards the config entry to the platforms?

Reply:
They are added as parrt of loading platforms, but this is to cope with entities that are detected while the system is already running and while the platform is already loaded. 

Let's say we have a bridge device with a light bulb. In the first instance we will call `async_load_platforms` which will look at the entity map, load the light platform, call add_listener, find a light bulb and add it. Here the `add_entities` in `async_refresh_entity_map` is pointless. As you say, when the platform is loaded it will find the light bulb and add it.

Now lets add a temperature sensor while the system is running. That will cause `c#` on the zeroconf metadata to increment which will trigger `async_refresh_entity_map` as well. In this case we haven't loaded the `sensor` platform yet. So this will take the same path as before and forward the config entry to the sensor platform. Again `add_entities` in `async_refresh_entity_map` is pointless.

However now I added another lightbulb to my system. HA discovery notices a change, `async_refresh_entity_map` is called. This time the light platform is already loaded. If I try to forward a config entry to light a second time it would fail as that config entry is already loaded on that platform. So now I need to check for new entities to add to platforms that are already active, and thats what the `add_entities` call is doing here.